app/services/course/assessment/answer/programming_auto_grading_service.rb

```diff
@@ -58,12 +59,12 @@ def evaluate_package(question, package)
   #   determine the correctness of the answer.
   # @return [Array<(Boolean, Integer, Course::Assessment::Answer::ProgrammingAutoGrading)>] The
   #   correct status, grade and the programming auto grading record.
-  def build_result(question, evaluation_result, ignore_evaluation:)
+  def build_result(question, evaluation_result, ignore_evaluation:, graded_test_case_types:)
     auto_grading = Course::Assessment::Answer::ProgrammingAutoGrading.new(actable: nil)
     set_auto_grading_results(auto_grading, evaluation_result)
     build_test_case_records(question, auto_grading, evaluation_result.test_reports)
-    test_results = auto_grading.test_results
-    test_cases = question.test_cases
+    test_cases = question.test_cases.where(test_case_type: graded_test_case_types)
+    test_results = auto_grading.test_results.reject { |r| graded_test_case_types.include?(r.test_case.test_case_type) }
```

Comment:
Is the `reject` condition reversed? Shouldn't the results for test case types which are in `graded_test_case_types` be included instead?

Reply:
oh gosh you're right. thanks for noticing!